cmd/argocd-application-controller/commands/argocd_application_controller.go

```diff
@@ -189,6 +191,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().BoolVar(&repoServerStrictTLS, "repo-server-strict-tls", env.ParseBoolFromEnv("ARGOCD_APPLICATION_CONTROLLER_REPO_SERVER_STRICT_TLS", false), "Whether to use strict validation of the TLS cert presented by the repo server")
 	command.Flags().StringSliceVar(&metricsAplicationLabels, "metrics-application-labels", []string{}, "List of Application labels that will be added to the argocd_application_labels metric")
 	command.Flags().StringVar(&otlpAddress, "otlp-address", env.StringFromEnv("ARGOCD_APPLICATION_CONTROLLER_OTLP_ADDRESS", ""), "OpenTelemetry collector address to send traces to")
+	command.Flags().StringSliceVar(&applicationNamespaces, "application-namespaces", []string{}, "List of additional namespaces that applications are allowed to be created in")
```

Comment:
Looks like this isn't making use of the `ARGOCD_APPLICATION_NAMESPACES` env var. Should it be?

Reply:
Maybe this setting should just live in `argocd-cm`, since it's shared by multiple components (the controller and the API)?

Reply:
I think we could make the env var `ARGOCD_APPLICATION_NAMESPACES` live in `argocd-cmd-params-cm` ConfigMap and have it consumed by both, argocd-server and argocd-application-controller in the Cobra commands instead of putting it in the `argocd-cm` ConfigMap.

A change to this setting would require restart of both, server and controller, so I think `argocd-cm` might not be the right place for it.

Reply:
Makes sense to me!